hello.py

```diff
@@ -386,180 +399,460 @@ def image_ref(self, repo):
+            args = args + ' --env=\'["%s"]\'' % env.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        
+        # if BenchRunner.CMD_ARG_WAIT[repo].waitline != "":
+        #     optimize_bin = "%s --wait-on-line=\"%s\"" % (optimize_bin, BenchRunner.CMD_ARG_WAIT[repo].waitline)
+        
+        cmd = ('%s record --period=%s --file=%s/%s.txt %s %s %s/%s:latest' %
+            (optimize_bin, period, accessed_files_dir, repo , entry, args, target, repo))
+        
+        return cmd
+    
+    def optimize_cmd_url_wait_cmd(self, repo, optimize_bin, accessed_files_dir, target, period):
+        entry = '-entrypoint \'["/bin/sh", "-c"]\''
+        args = ""
+        if BenchRunner.CMD_URL_WAIT[repo].arg != "":
+            args = args + '--args=\'["%s"]\'' % BenchRunner.CMD_URL_WAIT[repo].arg.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        if len(BenchRunner.CMD_URL_WAIT[repo].env) > 0:
+            env = " ".join(["-e %s=%s" % (k, v) for k, v in BenchRunner.CMD_URL_WAIT[repo].env.items()])
+            args = args + ' --env=\'["%s"]\'' % env.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        for a,b in BenchRunner.CMD_URL_WAIT[repo].mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            args += ' --mount type=bind,src=%s,dst=%s,options=rbind' % (a,b)
+            
+        cmd = ('%s record --period=%s --file=%s/%s.txt %s %s %s/%s:latest' %
+            (optimize_bin, period, accessed_files_dir, repo , entry, args, target, repo))
+        return cmd
+
+    def optimize(self, bench, optimize_bin, target_registry, target_repo) -> str:
+        accessed_files_dir = "accessed_files"
+        if not os.path.exists(accessed_files_dir):
+            os.mkdir(accessed_files_dir)
+        repo = image_repo(bench.name)
+        period = 30
+        
+        if repo in BenchRunner.ECHO_HELLO:
+            optimize_cmd = self.optimize_echo_hello_cmd(repo, optimize_bin, accessed_files_dir, target_registry + "/" + target_repo, period)
+            print(optimize_cmd)
+            rc = os.system(optimize_cmd)
+            return optimize_cmd
+        elif repo in BenchRunner.CMD_ARG:
+            optimize_cmd = self.optimize_cmd_arg_cmd(repo, optimize_bin, accessed_files_dir, target_registry + "/" + target_repo, period)
+            print(optimize_cmd)
+            rc = os.system(optimize_cmd)
+            return optimize_cmd
+        elif repo in BenchRunner.CMD_STDIN:
+            optimize_cmd = self.optimize_cmd_stdin_cmd(repo, optimize_bin, accessed_files_dir, target_registry + "/" + target_repo, period)
+            print(optimize_cmd)
+            
+            # p = subprocess.Popen(optimize_cmd, shell=True, stdin=subprocess.PIPE,
+            # stdout=sys.stdout, stderr=sys.stdout, bufsize=0)
+            # print(BenchRunner.CMD_STDIN[repo].stdin)
+            # stdin=BenchRunner.CMD_STDIN[repo].stdin+'\nexit\n'
+            # p.communicate(stdin.encode())
```

Comment:
Please remove dead code if they not needed in the future

Reply:
Done.